crates/scalajs/scala/wit.scala

```diff
@@ -0,0 +1,685 @@
+package object wit {
+
+  import scala.scalajs.js
+  import scala.scalajs.js.JSConverters._
+  import scala.scalajs.js.|
+  import scala.scalajs.js.annotation.JSName
+
+  sealed trait Nullable[+A] extends js.Any
```

Comment:
Given the Scala has non-nullable reference types by default, the consensus in Scala.js is to represent "nullable `A`" as `A`. So a `nullable String` is a `String`.

For primitives, we use their boxed variants : a `nullable Int` is a `java.lang.Integer`.

Reply:
This is added to not loose the WIT type information of something being `optional<>`. Even though Componentize.js decided to represent optionals as nulls we don't have to expose it like that for Scala, in my opinion.

Reply:
I disagree. The philosophy of Scala.js is to present JS as it is, without artifacts. Then, user-land libraries can build abstractions *on top* of the real thing.

Also, when targeting Scala 3, we can actually write `A | Null`, so that users using `-Yexplicit-nulls` can benefit from it all.

Reply:
I understand this philosophy of Scala.js - but in this case I'm not trying to present JS at all - I'm trying to present a component model interface which _has_ optionals. Having JS in between is just a temporary "annoyance" that, as you wrote earlier, will go away.

So in my opinion the users of this would like to define a WIT interface containing optionals, and implement it in Scala. How it is represented in JS is not relevant for them.

About targeting Scala 3, I've added a command line option to set the dialect (2 vs 3) but not really taking advantage of it yet, it can be added easily.